File: php/src/Auth/ASAPAuth.php
<?php

namespace Auth;

use Psr\Http\Message\ServerRequestInterface;

class ASAPAuth extends BaseAuth
{
    private string $issuer;
    private string $audience;
    private string $privateKey;

    public function __construct(string $issuer, string $audience, string $privateKey)
    {
        $this->type = AuthType::ASAP;
        $this->issuer = $issuer;
        $this->audience = $audience;
        $this->privateKey = $privateKey;
    }

    public function authenticate(ServerRequestInterface $request): bool
    {
        // TODO: Implement ASAP validation
        return true;
    }

    public function addAuth(ServerRequestInterface $request): ServerRequestInterface
    {
        // TODO: Add Authorization header
        return $request;
    }
} 

File: go/src/auth/asap_auth.go
package auth

import (
	"net/http"
)

// ASAPAuth implements the Authenticator interface for ASAP (Atlassian) Authentication
type ASAPAuth struct {
	BaseAuth
	Issuer     string
	Audience   string
	PrivateKey string
}

// NewASAPAuth creates a new ASAPAuth instance
func NewASAPAuth(issuer, audience, privateKey string) *ASAPAuth {
	return &ASAPAuth{
		BaseAuth:   BaseAuth{Type: ASAPAuth},
		Issuer:     issuer,
		Audience:   audience,
		PrivateKey: privateKey,
	}
}

// Authenticate validates the ASAP token
func (a *ASAPAuth) Authenticate(r *http.Request) error {
	// TODO: Implement ASAP validation
	return nil
}

// AddAuth adds ASAP token to the request
func (a *ASAPAuth) AddAuth(r *http.Request) {
	// TODO: Add Authorization header
} 